src/generator.rs

```diff
@@ -90,45 +90,161 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
     Ok(())
 }
 
+fn write_contents(output_directory: &Path, filename: &str, contents: &str) -> Result<()> {
+    let path = output_directory.join(filename);
+    make_parent(&path)?;
+
+    let contents = format!(
+        "\
+# Automatically generated by {exe_name:?}
+
+{contents}",
+        exe_name = std::env::current_exe().unwrap(),
+        contents = contents
+    );
+
+    fs::write(&path, contents).with_context(|| format!("Failed to write {:?}", path))
+}
+
 fn handle_device(output_directory: &Path, device: &Device) -> Result<u64> {
+    if device.is_swap() {
+        handle_zram_swap(output_directory, device)?;
+    } else {
+        handle_zram_mount_point(output_directory, device)?;
+    }
+
+    /* Return the device number */
+
+    device.name[4..]
+        .parse()
+        .with_context(|| format!("zram device \"{}\" number", device.name))
+}
+
+fn handle_zram_swap(output_directory: &Path, device: &Device) -> Result<()> {
     let swap_name = format!("dev-{}.swap", device.name);
+
     info!(
         "Creating unit {} (/dev/{} with {}MB)",
         swap_name,
         device.name,
         device.disksize / 1024 / 1024
     );
 
-    let swap_path = output_directory.join(&swap_name);
+    /* systemd-zram-setup@.service.
+     * We use the packaged unit, and only need to provide a small drop-in. */
+
+    write_contents(
+        output_directory,
+        &format!(
+            "systemd-zram-setup@{}.service.d/bindsto-swap.conf",
+            device.name
+        ),
+        &format!(
+            "\
+[Unit]
+BindsTo=dev-%i.swap
+"
+        ),
+    )?;
 
-    let contents = format!(
-        "\
-# Automatically generated by zram-generator
+    /* dev-zramX.swap */
 
+    write_contents(
+        output_directory,
+        &swap_name,
+        &format!(
+            "\
 [Unit]
-Description=Compressed swap on /dev/{zram_device}
+Description=Compressed Swap on /dev/{zram_device}
```

Comment:
Why the capitalisation change?

Reply:
Because the string is used as a title by systemd, non-capitalized looks strange in "Starting ...", "Stopping ...". I added an explanation to the commit message too.